src/napari_micromanager/_mda_handler.py

```diff
@@ -290,20 +396,29 @@ def _id_idx_layer(event: ActiveMDAEvent) -> tuple[str, tuple[int, ...], str]:
             - `layer_name` is the name of the corresponding layer in the viewer.
     """
     meta = cast("SequenceMeta", event.sequence.metadata.get(SEQUENCE_META_KEY))
-    axis_order = list(event.sequence.used_axes)
+
+    axis_order = _get_axis_labels(event.sequence)
 
     suffix = ""
     prefix = meta.file_name if meta.should_save else "Exp"
+
     if meta.split_channels and event.channel:
-        # Remove 'c' from idxs if we are splitting channels
-        # also prepare the channel suffix that we use for keeping track of arrays
         suffix = f"_{event.channel.config}_{event.index['c']:03d}"
         axis_order.remove("c")
 
     _id = f"{event.sequence.uid}{suffix}"
 
     # the index of this event in the full zarr array
-    im_idx = tuple(event.index[k] for k in axis_order)
+    im_idx: tuple[int, ...] = ()
+    for k in axis_order:
+        try:
+            im_idx += (event.index[k],)
+        # if axis not in event.index
+        # e.g. if we have bot a position sequence grid and a single position
```

Comment:
```suggestion
        # e.g. if we have both a position sequence grid and a single position
```

im not sure I fully understand this comment. Can you give an example of how to create a sequence that would trigger this?

Reply:
this is an example of a `MDASequence` that I found it needs something like this try/except:

```py
seq = MDASequence(
    channels=["DAPI"],
    stage_positions=[
        {"x": 0, "y": 0, "z": 0, "sequence": {"grid_plan": {"rows": 2, "columns": 2}}},
        {"x": 0, "y": 0, "z": 0}
    ],
)
```
In this case, the `g` index is present when iterating through the _first position_ but **not** in the _second position_ because it does not have a `grid_plan` sub-sequence.